pkg/p2p/p2p.go

```diff
@@ -26,6 +26,8 @@ type Service interface {
 	Peers() []Peer
 	SetNotifier(topology.Notifier)
 	Addresses() ([]ma.Multiaddr, error)
+	SetWelcomeMessage(val string) error
+	WelcomeMessageSynced() string
```

Comment:
I would ommit `Synced` in the name of the method name here. Also, is it really needed to expand the p2p layer with these methods?